File: sdk-go/examples/basic_usage.go
package main

import (
	"fmt"
	"log"

	"github.com/Masoud-Masoori/daena/sdk-go"
)

func main() {
	// Initialize client
	client := daena.NewClient(daena.ClientConfig{
		APIKey:  "your-api-key-here",
		BaseURL: "https://api.daena.ai",
	})

	// Test connection
	fmt.Println("Testing connection...")
	connected, err := client.TestConnection()
	if err != nil {
		log.Fatalf("Connection test failed: %v", err)
	}
	fmt.Printf("✅ Connected: %v\n\n", connected)

	// Get system health
	fmt.Println("Getting system health...")
	health, err := client.GetHealth()
	if err != nil {
		log.Fatalf("Failed to get health: %v", err)
	}
	fmt.Printf("✅ Status: %s\n", health.Status)
	if health.Version != "" {
		fmt.Printf("   Version: %s\n", health.Version)
	}
	fmt.Println()

	// Get system summary
	fmt.Println("Getting system summary...")
	summary, err := client.GetSystemSummary()
	if err != nil {
		log.Fatalf("Failed to get summary: %v", err)
	}
	fmt.Printf("✅ Total Agents: %d\n", summary.TotalAgents)
	fmt.Printf("   Active Agents: %d\n", summary.ActiveAgents)
	fmt.Printf("   Departments: %d\n", summary.Departments)
	fmt.Println()

	// Chat with Daena
	fmt.Println("Chatting with Daena...")
	chatResponse, err := client.Chat("Hello! What can you do?")
	if err != nil {
		log.Fatalf("Chat failed: %v", err)
	}
	fmt.Printf("✅ Response: %s\n", chatResponse.Response)
	fmt.Println()

	// Get all agents
	fmt.Println("Getting all agents...")
	agents, err := client.GetAgents("")
	if err != nil {
		log.Fatalf("Failed to get agents: %v", err)
	}
	fmt.Printf("✅ Found %d agents\n", len(agents))
	if len(agents) > 0 {
		fmt.Printf("   First agent: %s (%s)\n", agents[0].Name, agents[0].Department)
	}
	fmt.Println()

	// Store memory
	fmt.Println("Storing memory...")
	memoryReq := daena.StoreMemoryRequest{
		Key:       "example:test:data",
		ClassName: "example_data",
		Payload: map[string]interface{}{
			"message": "This is a test memory",
			"timestamp": "2025-01-XX",
		},
	}
	memory, err := client.StoreMemory(memoryReq)
	if err != nil {
		log.Fatalf("Failed to store memory: %v", err)
	}
	fmt.Printf("✅ Stored memory: %s\n", memory.RecordID)
	fmt.Printf("   Key: %s\n", memory.Key)
	fmt.Println()

	// Retrieve memory
	fmt.Println("Retrieving memory...")
	retrieved, err := client.RetrieveMemory("example:test:data")
	if err != nil {
		log.Fatalf("Failed to retrieve memory: %v", err)
	}
	fmt.Printf("✅ Retrieved: %s\n", retrieved.Key)
	fmt.Println()

	// Run council debate
	fmt.Println("Running council debate...")
	debateReq := daena.CouncilDebateRequest{
		Department: "product",
		Topic:      "Should we prioritize feature X?",
		Context: map[string]interface{}{
			"budget": 50000,
			"timeline": "Q1 2025",
		},
	}
	decision, err := client.RunCouncilDebate(debateReq)
	if err != nil {
		log.Fatalf("Council debate failed: %v", err)
	}
	fmt.Printf("✅ Decision: %s\n", decision.Decision)
	fmt.Printf("   Confidence: %.2f%%\n", decision.Confidence*100)
	fmt.Println()

	fmt.Println("✅ All operations completed successfully!")
}

File: sdk/examples/basic_usage.py
"""
Basic usage examples for Daena Python SDK.
"""

from daena_sdk import DaenaClient
import os

# Initialize client
API_KEY = os.getenv("DAENA_API_KEY", "test-api-key")
BASE_URL = os.getenv("DAENA_BASE_URL", "http://localhost:8000")

client = DaenaClient(api_key=API_KEY, base_url=BASE_URL)


def example_system_operations():
    """Example: System operations."""
    print("=" * 60)
    print("SYSTEM OPERATIONS")
    print("=" * 60)
    
    # Test connection
    if client.test_connection():
        print("✅ Connected to Daena!")
    else:
        print("❌ Connection failed")
        return
    
    # Get health
    health = client.get_health()
    print(f"Health Status: {health.get('status', 'unknown')}")
    
    # Get system summary
    summary = client.get_system_summary()
    print(f"Total Agents: {summary.get('agents', {}).get('total', 0)}")
    print(f"Departments: {summary.get('departments', {}).get('total', 0)}")
    
    # Get metrics
    metrics = client.get_system_metrics()
    print(f"Active Agents: {metrics.active_agents}")
    print(f"Average Latency: {metrics.average_latency_ms:.2f}ms")


def example_agent_management():
    """Example: Agent management."""
    print("\n" + "=" * 60)
    print("AGENT MANAGEMENT")
    print("=" * 60)
    
    # Get all agents
    agents = client.get_agents(limit=10)
    print(f"Total Agents: {len(agents)}")
    
    # Get agents by department
    if agents:
        first_agent = agents[0]
        print(f"First Agent: {first_agent.name} ({first_agent.department})")
        
        # Get specific agent details
        try:
            agent = client.get_agent(first_agent.agent_id)
            print(f"Agent Status: {agent.status}")
            print(f"Capabilities: {', '.join(agent.capabilities[:3])}")
        except Exception as e:
            print(f"Error getting agent: {e}")


def example_chat():
    """Example: Chat with Daena."""
    print("\n" + "=" * 60)
    print("DAENA CHAT")
    print("=" * 60)
    
    try:
        # Send a message
        response = client.chat("What's the current system status?")
        print(f"Response: {response.get('response', 'No response')[:200]}")
    except Exception as e:
        print(f"Chat error: {e}")


def example_memory_operations():
    """Example: Memory operations."""
    print("\n" + "=" * 60)
    print("MEMORY OPERATIONS")
    print("=" * 60)
    
    try:
        # Store memory
        record = client.store_memory(
            key="example:test:data",
            payload={
                "message": "This is a test memory",
                "timestamp": "2025-01-XX"
            },
            class_name="example"
        )
        print(f"✅ Stored memory: {record.record_id}")
        
        # Retrieve memory
        retrieved = client.retrieve_memory("example:test:data")
        if retrieved:
            print(f"✅ Retrieved memory: {retrieved.key}")
        
        # Search memory
        results = client.search_memory("test", limit=5)
        print(f"✅ Found {len(results)} matching memories")
        
    except Exception as e:
        print(f"Memory error: {e}")


def example_council_system():
    """Example: Council system."""
    print("\n" + "=" * 60)
    print("COUNCIL SYSTEM")
    print("=" * 60)
    
    try:
        # Run debate
        decision = client.run_council_debate(
            department="ai",
            topic="Should we optimize the memory system?",
            context={"priority": "high"}
        )
        print(f"✅ Decision: {decision.decision[:100]}")
        print(f"Confidence: {decision.confidence:.2%}")
        
        # Get conclusions
        conclusions = client.get_council_conclusions(limit=5)
        print(f"✅ Recent Conclusions: {len(conclusions)}")
        
    except Exception as e:
        print(f"Council error: {e}")


def example_analytics():
    """Example: Analytics."""
    print("\n" + "=" * 60)
    print("ANALYTICS")
    print("=" * 60)
    
    try:
        # Get analytics summary
        summary = client.get_analytics_summary()
        print(f"✅ Analytics Summary: {len(summary)} metrics")
        
        # Get insights
        insights = client.get_advanced_insights()
        print(f"✅ Advanced Insights: Available")
        
    except Exception as e:
        print(f"Analytics error: {e}")


if __name__ == "__main__":
    print("🚀 Daena Python SDK - Basic Usage Examples\n")
    
    try:
        example_system_operations()
        example_agent_management()
        example_chat()
        example_memory_operations()
        example_council_system()
        example_analytics()
        
        print("\n" + "=" * 60)
        print("✅ All examples completed!")
        print("=" * 60)
        
    except KeyboardInterrupt:
        print("\n\n⚠️  Interrupted by user")
    except Exception as e:
        print(f"\n\n❌ Error: {e}")
        import traceback
        traceback.print_exc()

